src/Altinn.Platform/Altinn.Platform.Storage/IntegrationTest/InstanceMultipartTests.cs

```diff
@@ -0,0 +1,180 @@
+                        {
+                            string filename = element.StorageUrl;
+                            string dataUrl = "/data/" + element.Id;
+
+                            string dataDeleteUrl = url + dataUrl;
+
+                            client.DeleteAsync(dataDeleteUrl);
+                        }
+                    }
+
+                    string instanceUrl = $"{versionPrefix}/instances/{instance.Id}?hard=true";
+                    client.DeleteAsync(instanceUrl);
+                }
+            }
+
+            DeleteApplicationMetadata();
+        }
+
+        private Application CreateTestApplication()
+        {
+            ApplicationClient appClient = new ApplicationClient(client);
+
+            try
+            {
+                Application existingApp = appClient.GetApplication(testAppId);
+                return existingApp;
+            }
+            catch (Exception)
+            {
+                // do nothing.
+            }
+
+            LanguageString title = new LanguageString
+            {
+                { "nb", "Testapplikasjon" },
+                { "en", "Test application" }
+            };
+
+            return appClient.CreateApplication(testAppId, title);
+        }
+
+        private Application DeleteApplicationMetadata()
+        {
+            ApplicationClient appClient = new ApplicationClient(client);
+
+            Application existingApp = appClient.DeleteApplication(testAppId);
+
+            return existingApp;
+        }
+
+        /// <summary>
+        /// Store a multipart file in one post operation
+        /// </summary>
+        [Fact]
+        public async void StoreMultiPartFileInOnePostOperation()
+        {
+            Instance instance = new Instance()
+            {
+                Id = Guid.NewGuid().ToString(),
+                InstanceOwnerId = "1000",
+                AppId = "tests/sailor",
+                Labels = new List<string>()
+                {
+                    "Hei"
+                },
+                DueDateTime = DateTime.Parse("2019-10-01")
+            };
+            
+            MultipartFormDataContent form = new MultipartFormDataContent();
+
+            form.Add(instance.AsJson(), "instance");
+            string xmlText = File.ReadAllText("data/example.xml");
+
+            form.Add(new StringContent(xmlText, Encoding.UTF8, "application/xml"), "default");
+        
+            string requestUri = $"{versionPrefix}/instances?appId={instance.AppId}&instanceOwnerId={instance.InstanceOwnerId}";
+
+            HttpResponseMessage response = await client.PostAsync(requestUri, form);
+            
+            response.EnsureSuccessStatusCode();
+
+            string result = await response.Content.ReadAsStringAsync();
+
+            Instance instanceResult = JsonConvert.DeserializeObject<Instance>(result);
+
+            // Assert
+            Assert.Equal("1000", instanceResult.InstanceOwnerId);
```

Comment:
Should also check that the data section is populated.

Reply:
Thanks, fixed.

Reply:
A bit more checking is required. For example, check that the data element contains correct expected elementType, contentType,  storage url, ...

Reply:
Fixed.